### ISSUE TITLE
Is this project still maintained?

### ISSUE THREAD

**Issue by Setheck** [k3s-io/k3s-ansible#220]
After submitting an issue over a year ago there has been no movement. 
https://github.com/k3s-io/k3s-ansible/issues/190 
I think missing scripts for basic management in alignment with k3s docs makes using this playbook unusable.

Is this project still maintained?

**Comment by KI7ODK** [comment]
I was wondering the same. There are multiple issues that have been posted which I believe are related to the same simple fix. No progress has been made.

**Comment by ankathur** [comment]
+1

**Comment by azerbe** [comment]
+1

**Comment by LarssonOliver** [comment]
It's really sad to see this project like this. It was in a good state and there seems to be people willing to contribute to keep it up to date. Might be worth trying to get in contact with someone at Rancher and the k3s team about this. They did take the project in under their org once upon a time after all.

**Comment by tuupola** [comment]
As a sidenote I have had good results with this:

https://github.com/PyratLabs/ansible-role-k3s

Although it is also looking for new maintainer.

**Comment by KI7ODK** [comment]
> It's really sad to see this project like this. It was in a good state and there seems to be people willing to contribute to keep it up to date. Might be worth trying to get in contact with someone at Rancher and the k3s team about this. They did take the project in under their org once upon a time after all.

I was thinking along the same lines.

**Comment by St0rmingBr4in** [comment]
Reading this proves me that I cleariy don't have the time to maintain this anymore.
Unfortunately I don't have the rights to give maintainer permissions.

I would say we have a couple of options:

1. Find at least 2 people to help @itwars and me to run this. Then ask one of k3s org members to add them as maintainer.
2. Deprecate this in favor of https://github.com/PyratLabs/ansible-role-k3s and join forces

About 2. I know that quite a few people like the simplicity of k3s-ansible compared to ansible-role-k3s. I didn't feel that when I tried it.

What do you think about 1 or 2 ?

**Comment by stafwag** [comment]
As a contributor to this project;

I started to use https://github.com/PyratLabs/ansible-role-k3s the author of the role asked to merge the projects a couple of months back ( not sure if this issue is still open ) because he wanted/needed some help. He got some help to maintain his project now if I understood it correctly

While I like the simplicity and the code in this project. 
I like https://github.com/PyratLabs/ansible-role-k3s a lot. The fact that it's a "real" role also makes it easier to integrate in other projects.

So I'd go with option 2 looks if we reuse the code from this project.

**Comment by stafwag** [comment]
> As a sidenote I have had good results with this:
> 
> https://github.com/PyratLabs/ansible-role-k3s
> 
> Although it is also looking for new maintainer.

He got some help now if I understood it correctly...

**Comment by drewburr** [comment]
I agree with the sentiment of having an importable role. It feels odd to clone a repo to run a playbook. ansible-role-k3s has better alignment with Ansible's code sharing practices. Plus, consolidating down to one repo should also increase overall activity, producing a better product.

In terms of consumability, I feel that with some TLC on the docs, this would be much easier for someone starting off. The README would benefit greatly from some reorganization and reformatting.

**Comment by dereknola** [comment]
As part of Hackweek and because I have the permissions as a K3s dev, I am attempting to cleanup this repo as much as possible. That includes:
- HA (Wrangling the 3-4 different PRs that have implemented this over the years into 1)
- Better README instructions (Also point people to other Ansible projects for K3s)
- Upgrade playbook
- Merging as many PRs as possible and closing Issues.


There likely won't be an expansion of utility that people would want. So likely no:
- External load balancers
- Private registries
- Longhorn

Basically, it will utilize everything K3s ships by default and not vary from that. There are many other ansible projects out there that expand  with K3s + Extra. And pointing those cool projects out should be a part of this repo. 

Afterwards I will attempt to maintain this repo, but obviously people have forked and largely moved on from this project.

**Comment by LarssonOliver** [comment]
Really happy to see this come to life again. Kudos to @dereknola!

**Comment by domvo** [comment]
Big thanks also from my side. Really happy to see this is maintained.

**Comment by sheerun** [comment]
What is maintained?

**Comment by bbaassssiiee** [comment]
Yes, 11 PRs merged in 2025 alone. IMHO this issue can be closed.
